music21/analysis/enharmonics.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+#-------------------------------------------------------------------------------
+# Name:         enharmonics.py
+# Purpose:      Tools for returning best enharmonics
+#
+# Authors:      Mark Gotham
+#               Michael Scott Cuthbert
+#
+# Copyright:    Copyright © 2017 Michael Scott Cuthbert and the music21 Project
+# License:      LGPL or BSD, see license.txt
+#-------------------------------------------------------------------------------
+
+import unittest
+import itertools
+from music21 import common
+from music21 import exceptions21
+from music21 import pitch
+from music21 import chord
+from music21 import musedata
+
+from music21 import environment
+_MOD = 'analysis.enharmonics'
+environLocal = environment.Environment(_MOD)
+
+
+class EnharmonicsException(exceptions21.Music21Exception):
+    pass
+
+class EnharmonicScoreRules:
+    def __init__(self):
+        self.sameStaffLine = False
+        self.alterationPenalty = 4
+        self.augDimPenalty = 2
+        self.mixSharpsFlatsPenalty = False
+
+class ChordEnharmonicScoreRules(EnharmonicScoreRules):
+    def __init__(self):
+        super().__init__()
+        self.mixSharpsFlatsPenalty = 2
+
+class EnharmonicSimplifier:
+    '''
+    Takes any pitch list input and returns the best enharmonic respelling according to the input
+    criteria and rule weightings.
+    Those criteria and rule weightings are currently fixed, but in future the user should be able
+    to select their own combination and weighting of rules according to preferences,
+    with predefined defaults for melodic and harmonic norms.
+    Note: EnharmonicSimplifier itself returns nothing.
+    '''
+    def __init__(self, pitchList, ruleClass=EnharmonicScoreRules):
```

Comment:
For our documentation system, we need for all classes to be instantiable with no required arguments, so just set pitchList=None  (I'll double check that this is still true, but it might be)